src/torchcodec/decoders/_core/VideoDecoder.cpp

```diff
@@ -1606,6 +1669,56 @@ void VideoDecoder::createSwsContext(
   streamInfo.swsContext.reset(swsContext);
 }
 
+void VideoDecoder::createSwrContext(
+    StreamInfo& streamInfo,
+    int sampleRate,
+    AVSampleFormat sourceSampleFormat,
+    AVSampleFormat desiredSampleFormat) {
+  SwrContext* swrContext = nullptr;
+
+  int status = AVSUCCESS;
```

Comment:
Below: similar comment, is it worth having a separate function in `FFmpegCommon.h` to do the initializaton?

Reply:
Yes, for the same reasons.